goodmap/data_validator.py

```diff
@@ -0,0 +1,61 @@
+import json
+from sys import argv
+
+
+def check_json_validaty(json_file):
+    json_data = json.load(json_file)
+    return json_data
+
+
+def check_obligatory_fields(datapoints, obligatory_fields):
+    for p in datapoints:
+        for field in obligatory_fields:
+            if field not in p.keys():
+                print(f'ERROR, datapoint: \n {p} \n is missing obligatory field: "{field}"')
+
+
+def check_categories_values(datapoints, categories):
+    for p in datapoints:
+        for category in categories & p.keys():
+            if type(p[category]) is list:
+                for attribute_value in p[category]:
+                    if attribute_value not in categories[category]:
+                        print(
+                            f'ERROR, datapoint: \n {p} \n \
+                             has an invalid value for category: "{category}"'
+                        )
+            else:
+                if p[category] not in categories[category]:
+                    print(
+                        f'ERROR, datapoint: \n {p} \n \
+                             has an invalid value for category: "{category}"'
+                    )
+
+
+def check_for_null_values(datapoints):
+    for p in datapoints:
+        ### check for null values in non-obligatory fields
+        for attribute in p.keys():
+            if p[attribute] is None:
```

Comment:
```suggestion
        for value in p.values():
            if value is None:
```

Reply:
This is visible in multiple places.

Reply:
I need to iterate over attributes because I print (add to message) the name of the faulty attribute.

Reply:
You're right. sorry.

```suggestion
        for attribute, value in p.items():
            if value is None:
```

then